sesman/session.c

```diff
@@ -374,7 +374,9 @@ session_get_avail_display_from_chain(void)
 
     display = g_cfg->sess.x11_display_offset;
 
-    while ((display - g_cfg->sess.x11_display_offset) <= g_cfg->sess.max_sessions)
+    /* positive max_sessions value limits the max number of sessions */
+    while (g_cfg->sess.max_sessions <= 0 ||
+           (display - g_cfg->sess.x11_display_offset) <= g_cfg->sess.max_sessions)
```

Comment:
It would be nice to have a limit on the number of tries to prevent xrdp from getting stuck in an infinite loop if things go wrong (no network support in kernel, exhausted limit of connections etc). But the logic should be different here from what it is now. We can expect that every existing session used a display number. We can accept that a certain number of display numbers (say, 100) are unavailable for other reasons. Once the number of tries exceeds the number of existing sessions plus that fudge factor, the code should fail.

The maximal number of sessions should not be used here.